src/XrdVoms/XrdVomsMapfile.cc

```diff
@@ -367,8 +413,70 @@ XrdVomsMapfile::Configure(XrdSysError *erp, XrdHttpSecXtractor *xtractor)
     if (!map_filename.empty()) {
         if (erp) erp->Emsg("Config", "Will initialize VOMS mapfile", map_filename.c_str());
         mapper.reset(new XrdVomsMapfile(erp, xtractor, map_filename));
-        mapper->Reconfigure();
     }
 
-    return mapper.get();
+    return mapper->IsValid() ? mapper.get() : nullptr;
+}
+
+
+void *
+XrdVomsMapfile::MaintenanceThread(void *myself_raw)
+{
+    auto myself = static_cast<XrdVomsMapfile*>(myself_raw);
+
+   auto now = monotonic_time_s();
+   auto next_update = now + m_update_interval;
+   while (true) {
+       now = monotonic_time_s();
+       auto remaining = next_update - now;
+       struct pollfd fds;
+       fds.fd = myself->m_maintenance_pipe_r;
```

Comment:
My take is this is only for the case when the object is destroyed and you want to stop the maintenance thread. OK, I get that but the object will never be destroyed as far as I can see. Once the mapfile is started it continues for the life of the server. So, I am confused as to what problem this is trying to solve.

Reply:
It is a bit overkill - I only triggered it during the destruction of the object when the startup configuration failed.

I figured I'd err toward meticulous cleanup (the inspiration is in the temp CA handler code) instead of just assuming nothing needs to be torn down.